FIX: Add message for single column in associations tab

mithilP007 opened the pull request:
Fixes #1747

Adds a conditional check to display a clear message when there is only one column in the dataframe, instead of showing the misleading "No strong associations" message.

The new message: "No associations computed because there is only one column."

mithilP007 committed:
FIX: Add message for single column in associations tab

Fixes #1747

Adds a conditional check to display a clear message when there is only one column in the dataframe, instead of showing the misleading "No strong associations" message.

The new message: "No associations computed because there is only one column."

mithilP007 commented:
The test failure on `test (macos-latest, ci-py309-min-deps)` is due to a CI infrastructure issue, not related to the code changes in this PR.

The error shows a network timeout while fetching dependencies:
```
× failed to solve the pypi requirements of 'ci-nightly-deps'
└─► operation timed out
```

This is a transient issue with accessing `https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/pyarrow/`. All other 23 checks passed successfully. The code changes are working as intended.

rcap107 commented:
> The test failure on `test (macos-latest, ci-py309-min-deps)` is due to a CI infrastructure issue, not related to the code changes in this PR.
> 
> The error shows a network timeout while fetching dependencies:
> 
> ```
> × failed to solve the pypi requirements of 'ci-nightly-deps'
> └─► operation timed out
> ```
> 
> This is a transient issue with accessing `https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/pyarrow/`. All other 23 checks passed successfully. The code changes are working as intended.

I restarted the jobs, everything seems to be working now

rcap107 committed:
Merge remote-tracking branch 'upstream/main' into pr/mithilP007/1758

mithilP007 committed:
FIX: Remove inconsistent # noqa: F401 comments for consistency

Removed # noqa: F401 lint suppression comments from the else block (sklearn >= 1.5) to maintain consistency with the if block (sklearn < 1.5) where these comments were already removed. This addresses the code review feedback from coderabbital bot.

mithilP007 commented:
No
